### ISSUE TITLE
Feature request: Recreate backup from export

### ISSUE THREAD

**Issue by neilmusgrove** [xeals/signal-back#44]
This project looks great for exporting signal backups and I notice there is a raw option which worked for me. What I would find really useful is if I could export to raw formats (with all attachments), edit (manually), and then rebuild into a valid achive for import. Essentially all I think I need is to decrypt the bundle as is to disk, edit and then re-encrypt (eg no format conversion). I've got some broken data which I'm trying to fix and I'd like to have a go at fixing this manually in the sql export file and importing that back into signal with all attachments. Many Thanks.

**Comment by xeals** [comment]
This could be a bit of a long-term goal, but would require some work. Interesting proposal.

**Comment by 4lex11s** [comment]
Yes it should also useful, for importing data in Signal from another messaging app, such as [slack](https://github.com/Karumi/hangout-slack-import/blob/master/hangout-to-csv-slack.py) or google hangout. Many Thanks !

**Comment by alexlance** [comment]
If an XML file that was compatible with SMS Backup & Restore, were able to be rolled up into a file that could be imported into Signal, it would fill a gap for those wishing to migrate their existing message stores to Signal.

There is currently no method to migrate existing messages *inclusive of mms* into Signal.

**Comment by codethief** [comment]
Another benefit would be that backups could possibly be merged by hand. I still have an old phone lying around that has tens of thousands of Signal messages on it from a time when encrypted backups weren't possible yet. I would love to have these available on my current phone.

**Comment by samizdis** [comment]
Another benefit of this would be the ability to recover backups where some of the frames are corrupt, see https://github.com/signalapp/Signal-Android/issues/8355

Signal simply refuses to load partially-corrupt backup files, whereas signal-back will happily read/recover all it can.

**Comment by alexlance** [comment]
Far out it'd be nice to see some movement on this task.
Is it quite a difficult thing to implement?

**Comment by luke-shields** [comment]
This would also be very helpful for Huawei devices, most of which Signal is unable to import Sms from

**Comment by alexlance** [comment]
Any chance this is on the radar?
 
Would love to be able to dump out my old SMS message store to XML (using SMS Backup and Restore) and then use signal-back to convert that XML file into a Signal Backup file, that can then be restored into Signal complete with MMS/attachments.
